ex5/ex5.c

```diff
@@ -15,7 +15,34 @@ char* msg3 = "hello world #3";
 
 int main()
 {
-    // Your code here
-    
-    return 0;
+  // Your code here
+  char inbuf[MSGSIZE];
+  int p[2]; 
+
+  if (pipe(p) < 0)
+  {
+      fprintf(stderr, "pipe failed\n");
+      exit(1);
+  }
+  int c = fork();
+  if (c > 0)
+  {
+      printf("Child is writing\n");
+      write(p[1], msg1, MSGSIZE);
+      write(p[1], msg2, MSGSIZE);
+      write(p[1], msg3, MSGSIZE);
+  }
+  else
+  {
+      close(p[1]);
+      printf("Parent is reading\n");
+
+      for (int i = 0; i < 3; i++)
+      {
+          read(p[0], inbuf, MSGSIZE);
+          printf("%s\n", inbuf);
+      }
+
+      return 0;
+  }
 }
```

Comment:
Your exercises look great so far, keep up the good work! For exercise 6 you will be writing to null. There has been some confusion about this so I will give you a line of code that might clarify the write() function. Use: write(STDOUT_FILENO, NULL, sizeof(NULL));